### ISSUE TITLE
REST API

### ISSUE THREAD

**Issue by ShadowPeo** [garysharp/Disco#146]
From what we can see, the current API is limited when integrating with other software. Having a REST API would allow us to better integrate the system within our workflows

**Comment by garysharp** [comment]
You're correct in identifying that the API Disco ICT exposes is largely suited to support its own UI.
Authoring a separate API to cover all of Disco ICT's functionality would be initially prohibitive.
Perhaps we can start small, with a specific use-case, and make sure appropriate API's are available to meet those needs.
Did you have an idea of a workflow you would like to support? Feel free @ShadowPeo to reach out directly if you'd rather the conversation continue privately.

**Comment by ShadowPeo** [comment]
Posting this for someone else on the team

----------------------

**Comment by garysharp** [comment]
To read data from Disco ICT, I'd recommend connecting directly to the SQL database. Read-only permission can be assigned to either a user, or a SQL account. It is true that changes to the SQL schema could change over time (years) although these have proven to be largely additive (which don't break anything sensible).
The database is quite logically laid out. But definitely reach out if you need assistance locating some data.

**Comment by garysharp** [comment]
A reliable API is required to **create jobs**. This would be a restful API that accepted a json model and returned basic job details or relevant validation errors (in json format).

In the interim, the following PowerShell sample can be used to create a job with automation:
[JobCreationTemporaryAPI.txt](https://github.com/user-attachments/files/17898720/JobCreationTemporaryAPI.txt)

This method will be immediately obsolete (and unsupported) once the restful API to create jobs is delivered.

**Comment by SHSC-JMS** [comment]
There have been a heap of times I would have loved API integration to Disco.... Most are listed above, but my preferences would be:
1) For a given Serial Number return assignment details
2) For a given USERID return device Assignments
3) Return list of Devices ALA Export Devices functionality...
